src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -844,7 +844,7 @@ private string ReadName(int position, int nameLength)
                 if (checkErrors)
                 {
                     long result;
-                    while (Interop.CheckIo(result = sysCall(_inotifyHandle, arg1, arg2), isDirectory: true)) ;
+                    Interop.CheckIo(result = sysCall(_inotifyHandle, arg1, arg2), isDirectory: true);
```

Comment:
Is this SysCall function still needed?  ```checkErrors``` is passed in as a hardcoded value from the call site, which means the caller knows whether they want the checked or non-checked version, and each of those is now basically one line.  We could avoid the extra delegate allocations and invocations if this were to just go away.